includes/Core/Modules/Module.php

```diff
@@ -364,42 +387,52 @@ final protected function execute_data_request( Data_Request $data ) {
 	 * Validates the given data request.
 	 *
 	 * @since 1.9.0
+	 * @since n.e.x.t Requires `Datapoint` instead of `Data_Request`.
 	 *
-	 * @param Data_Request $data Data request object.
+	 * @param Datapoint $datapoint Datapoint instance.
+	 * @throws Insufficient_Scopes_Exception Thrown if the user has not granted.
+	 */
+	private function validate_data_request( Datapoint $datapoint ) {
+		// Validate the request scopes if it requires a service.
+		if ( $this instanceof Module_With_Scopes && $datapoint->get_service() ) {
```

Comment:
Can we really just add this wrapper? The scopes may still need to be validated even if there is no service? What about our closures that issue API requests? We spoke about changing them, but it still may introduce problems not validating this elsewhere. Most importantly, I don't see what benefit that brings us.